jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/upload/EditLambdaDialog.kt

```diff
@@ -0,0 +1,363 @@
+import software.aws.toolkits.jetbrains.core.AwsClientManager
+import software.aws.toolkits.jetbrains.core.credentials.ProjectAccountSettingsManager
+import software.aws.toolkits.jetbrains.services.iam.CreateIamRoleDialog
+import software.aws.toolkits.jetbrains.services.iam.IamRole
+import software.aws.toolkits.jetbrains.services.iam.listRolesFilter
+import software.aws.toolkits.jetbrains.services.lambda.Lambda.findPsiElementsForHandler
+import software.aws.toolkits.jetbrains.services.lambda.LambdaFunction
+import software.aws.toolkits.jetbrains.services.lambda.LambdaPackager
+import software.aws.toolkits.jetbrains.services.lambda.runtimeGroup
+import software.aws.toolkits.jetbrains.utils.notifyError
+import software.aws.toolkits.jetbrains.utils.notifyInfo
+import software.aws.toolkits.jetbrains.utils.ui.addAndSelectValue
+import software.aws.toolkits.jetbrains.utils.ui.blankAsNull
+import software.aws.toolkits.jetbrains.utils.ui.populateValues
+import software.aws.toolkits.jetbrains.utils.ui.selected
+import software.aws.toolkits.resources.message
+import java.awt.event.ActionEvent
+import java.util.concurrent.TimeUnit
+import java.util.function.Function
+import javax.swing.Action
+import javax.swing.JComponent
+
+private val DEFAULT_TIMEOUT = TimeUnit.MINUTES.toSeconds(1).toInt()
+private val MAX_TIMEOUT = TimeUnit.MINUTES.toSeconds(15).toInt()
+private const val MIN_TIMEOUT = 1
+
+class EditLambdaDialog(
+    private val project: Project,
+    private val isUpdate: Boolean,
+    name: String = "",
+    description: String = "",
+    runtime: Runtime? = null,
+    handlerName: String = "",
+    envVariables: Map<String, String> = emptyMap(),
+    timeout: Int = DEFAULT_TIMEOUT,
+    role: IamRole? = null
+) : DialogWrapper(project) {
+
+    constructor(project: Project, lambdaFunction: LambdaFunction) :
+            this(
+                project = project,
+                isUpdate = true,
+                name = lambdaFunction.name,
+                description = lambdaFunction.description ?: "",
+                runtime = lambdaFunction.runtime,
+                handlerName = lambdaFunction.handler,
+                envVariables = lambdaFunction.envVariables ?: emptyMap(),
+                role = lambdaFunction.role
+            )
+
+    private val view = EditLambdaPanel(project)
+    private val validator = UploadToLambdaValidator()
+    private val s3Client: S3Client = AwsClientManager.getInstance(project).getClient()
+    private val iamClient: IamClient = AwsClientManager.getInstance(project).getClient()
+
+    private val updateAction = UpdateLambdaSettingsAction()
+    private val deployAction = DeployLambdaAction()
+
+    private var validateDeploySettings = false
+
+    init {
+        super.init()
+        title = if (isUpdate) message("lambda.upload.edit.title", name) else message("lambda.upload.create.title")
+
+        view.name.text = name
+
+        if (isUpdate) {
+            view.name.isEnabled = false
```

Comment:
So we can't change the name of an existing function?

Reply:
Nope, UpdateFunction takes the name (or arn.....) in the name field, so rename isn't possible

https://docs.aws.amazon.com/cli/latest/reference/lambda/update-function-configuration.html

Reply:
cool